src/parser.js

```diff
@@ -338,17 +253,41 @@ define([
         }
         mug = adapt(mug, form);
         var node = form.tree.getNodeFromMug(mug);
-        if (node && node.parent.value !== parentMug) {
+        if (!node) {
+            // insert control-only mug into the tree
+            mug.options.isControlOnly = true; // TODO should not be mutating mug.options, check if this is necessary
+            node = form.tree.insertMug(mug, 'into', parentMug);
+            // HACK fix abstraction broken by direct tree insert
+            form.mugMap[mug.ufid] = mug;
+        } else if (node.parent.value !== parentMug) {
             mug.p.dataParent = node.parent.getAbsolutePath();
-            form.tree.insertMug(mug, 'into', parentMug);
+            node = form.tree.insertMug(mug, 'into', parentMug);
         }
         if (appearance) {
             mug.p.appearance = appearance;
         }
+
         if (!adapt.skipPopulate) {
-            populateMug(form, mug, $cEl, parentMug);
+            form.vellum.populateControlMug(mug, $cEl);
+
+            // add any arbitrary attributes that were directly on the control
+            mug.p.rawControlAttributes = getAttributes($cEl);
+        }
+        return node;
+    }
+
+    function populateControlMug(mug, $cEl) {
+        var labelEl = $cEl.children('label'),
+            hintEl = $cEl.children('hint');
+        if (labelEl.length && mug.spec.label.presence !== 'notallowed') {
+            var labelVal = xml.humanize(labelEl);
+            if (labelVal) {
+                mug.p.label = labelVal;
+            }
+        }
+        if (hintEl.length && mug.spec.hintLabel.presence !== 'notallowed') {
+            mug.p.hintLabel = xml.humanize(hintEl);
```

Comment:
Non-itext label parsing is done here, and itext label parsing is done in the javaRosa plugin.